src/overlays/actors/ovl_En_Slime/z_en_slime.c

```diff
@@ -45,146 +77,1170 @@ ActorInit En_Slime_InitVars = {
+    EnSlime_SetupInitializeIdle(this);
+}
+
+void EnSlime_Destroy(Actor* thisx, PlayState* play) {
+    EnSlime* this = THIS;
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+/**
+ * Upon immobilization via ice arrow (for chuchu types outside of blue),
+ * initialize ice visual effects before chuchu enters immobile state.
+ */
+void EnSlime_Freeze(EnSlime* this) {
+    this->drawDmgEffType = ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX;
+    this->drawDmgEffScale = 0.4f;
+    this->collider.base.colType = COLTYPE_HIT3;
+    this->drawDmgEffFrozenSteamScale = 0.6f;
+    this->drawDmgEffAlpha = 1.0f;
+    this->timer = 80;
+    this->actor.flags &= ~ACTOR_FLAG_400;
+    Actor_SetColorFilter(&this->actor, COLORFILTER_COLORFLAG_RED, 255, COLORFILTER_BUFFLAG_XLU, 80);
+}
+
+/**
+ * If stunned ice effect is active (ie. non-blue chuchu was frozen by ice arrow),
+ * initialize thaw and spawn thawing effect.
+ */
+void EnSlime_Thaw(EnSlime* this, PlayState* play) {
+    if (this->drawDmgEffType == ACTOR_DRAW_DMGEFF_FROZEN_NO_SFX) {
+        this->drawDmgEffType = 0; // So it's not triggered again until Freeze has been called again.
+        this->collider.base.colType = COLTYPE_NONE;
+        this->drawDmgEffAlpha = 0.0f;
+        Actor_SpawnIceEffects(play, &this->actor, this->limbPos, EN_SLIME_LIMBPOS_COUNT, 2, 0.2f, 0.2f);
+        this->actor.flags |= ACTOR_FLAG_200;
+    }
+}
+
+/**
+ * If blink animation is in progress, advance to next frame.
+ * If eyes are open, randomly (5% chance) initialize new blink cycle.
+ */
+void EnSlime_Blink(EnSlime* this) {
+    if (this->eyeTexIndex != EN_SLIME_EYETEX_OPEN) {
+        this->eyeTexIndex++;
+        if (this->eyeTexIndex == EN_SLIME_EYETEX_MAX) {
+            this->eyeTexIndex = EN_SLIME_EYETEX_OPEN;
+        }
+    } else if (Rand_ZeroOne() < 0.05f) {
+        this->eyeTexIndex = EN_SLIME_EYETEX_HALF;
+    }
+}
+
+/**
+ * Set the actor's initial scale and action function.
+ */
+void EnSlime_SetupInitializeIdle(EnSlime* this) {
+    this->actor.scale.x = 0.008f;
+    this->actor.scale.z = 0.008f;
+    this->actor.scale.y = 0.011f;
+    this->actionFunc = EnSlime_InitializeIdle;
+}
+
+/**
+ * Once actor touches the ground, it starts its idle behavior.
+ * This function is only called right after the actor is first initialized.
+ * It is not set as the action function again.
+ */
+void EnSlime_InitializeIdle(EnSlime* this, PlayState* play) {
+    if (this->actor.bgCheckFlags & BGCHECKFLAG_GROUND) {
+        this->actor.flags &= ~ACTOR_FLAG_10;
+        EnSlime_SetupIdle(this);
+    }
+}
+
+/**
+ * Set or restore idle action state.
+ * Sets timer to 25 and cycles to SitIdle action.
+ */
+void EnSlime_SetupIdle(EnSlime* this) {
+    this->timer = 25;
+    this->actor.speed = 0.0f;
+    this->actionFunc = EnSlime_Idle;
+}
+
+/**
+ * Sit idle for one frame, updating rotation and scale to give
+ * the actor an amorphous, oozing effect.
+ * Chuchu will engage player during idle sit if player is visible and
```

Comment:
I'd just get rid of "idle sit" from here; the comment still makes sense without it.